theano/gpuarray/multinomial.py

```diff
@@ -327,6 +327,7 @@ def gpu_kernels(self, node, name):
 
                     if (! %(replace)s )
                     {
+                        __syncthreads();
```

Comment:
A safer way in the short term may be to remove the parallel normalization, which would make sure only one thread reads / writes from each global location.